R/zchunk_LA124.heat.R

```diff
@@ -36,74 +37,189 @@ module_energy_LA124.heat_DISABLED <- function(command, ...) {
     A24.globaltech_coef <- get_data(all_data, "energy/A24.globaltech_coef")
     calibrated_techs <- get_data(all_data, "energy/calibrated_techs")
     enduse_fuel_aggregation <- get_data(all_data, "energy/enduse_fuel_aggregation")
-    L1011.en_bal_EJ_R_Si_Fi_Yh <- get_data(all_data, "temp-data-inject/L1011.en_bal_EJ_R_Si_Fi_Yh")
-    L1231.out_EJ_R_elec_F_tech_Yh <- get_data(all_data, "temp-data-inject/L1231.out_EJ_R_elec_F_tech_Yh")
+
+    get_data(all_data, "temp-data-inject/L1011.en_bal_EJ_R_Si_Fi_Yh") %>%
+      gather(year, value, -fuel, -sector, -GCAM_region_ID) %>%
+      mutate(year = sub("X", "", year)) %>%
+      mutate(year = as.integer(year)) -> L1011.en_bal_EJ_R_Si_Fi_Yh
+
+    get_data(all_data, "temp-data-inject/L1231.out_EJ_R_elec_F_tech_Yh") %>%
+      gather(year, value, -fuel, -sector, -GCAM_region_ID, -technology) %>%
+      mutate(year = sub("X", "", year)) %>%
+      mutate(year = as.integer(year)) -> L1231.out_EJ_R_elec_F_tech_Yh
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Create list of heat regions
+    A_regions %>%
+      filter(heat == 1) %>%
+      select(GCAM_region_ID) -> heat_regionIDs
+
+    # Fuel inputs to district heat
+    # Process fuel inputs in all regions; some will have the energy assigned to bld/ind, and others have a district heat sector
+    L1011.en_bal_EJ_R_Si_Fi_Yh  %>%
+      filter(sector == "in_heat") %>%
+      mutate(sector = sub("in_", "", sector)) %>%
+      left_join(enduse_fuel_aggregation, by="fuel") %>%
+      select(GCAM_region_ID, sector, year, value, heat) %>%
+      rename(fuel = heat) %>%
+      group_by(fuel, sector, GCAM_region_ID, year) -> temp %>%
```

Comment:
This line is syntactically incorrect, which means you didn't run the driver, much less the tests, before committing this and pushing it to GitHub. 15 demerits @jhoring !  😠   ...I have some work to do on this PR and will fix.